connector/google/google.go

```diff
@@ -0,0 +1,266 @@
+// Package google implements logging in through Google's OpenID Connect provider.
+package google
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/coreos/go-oidc"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/admin/directory/v1"
+
+	"github.com/coreos/dex/connector"
+)
+
+const (
+	issuerURL = "https://accounts.google.com"
+)
+
+// Config holds configuration options for Google logins.
+type Config struct {
+	ClientID     string `json:"clientID"`
+	ClientSecret string `json:"clientSecret"`
```

Comment:
nit: can this be optionally read from a file? e.g. `ClientSecretFile`

Reply:
I don't see this being done in any of the other connectors but could add this. I assume the `ClientSecretFile` would just contain a single string? Or is there a format for files from Google containing the `ClientSecret`?